keps/sig-cli/2985-public-krm-functions-registry/README.md

```diff
@@ -0,0 +1,1344 @@
+Catalog resource.
+
+```yaml
+apiVersion: config.k8s.io/v1alpha1
+kind: KRMFunction
+metadata:
+  # 'name' must match the last part of the image name for container-based function.
+  # for example: the name should be set-namespace when image name is docker.example.co/functions/set-namespace
+  name: set-namespace
+spec:
+  apiVersion: example.com/v1beta1 # Primary apiVersion
+  kind: SetNamespace # Primary kind
+  description: "A short description of the KRM function"
+  publisher: example.com
+  schema: <a URL pointing to the schema of the functionConfig of apiVersion example.com/v1beta1 and kind SetNamespace>
+  version: v1.2.3
+  idempotent: true|false
+  runtime:
+    container:
+      image: docker.example.co/functions/set-namespace:v1.2.3 # The version part should match `spec.version`.
+      sha256: a428de... # The digest of the image which can be verified against. This field is required if the version is semver.
+      requireNetwork: true|false
+      requireVolumeMount: true|false
+  additionalAPIVersionKinds:
+    - apiVersion: example.com/v1alpha1
+      kind: LegacySetNamespace
+      deprecated: true
+  usage: <a URL pointing to a README.md>
+  home: <a URL pointing to the home page>
+  examples:
+    - <a URL pointing to a README.md>
+    - <another URL pointing to another README.md>
+  license: Apache 2.0
+  maintainers: # The maintainers for this function. It doesn't need to be the same as the publisher OWNERS. 
+    - foo@example.com
+  tags: # keywords of the KRM functions
+    - mutator
+    - namespace
+```
+
+The following is an example for exec-based KRM function. We will not allow
+contributors to publish exec-based KRM functions. But we want to standardize the
+metadata to allow an organization to share exec-based KRM functions internally.
+
+```yaml
+apiVersion: config.k8s.io/v1alpha1
+kind: KRMFunction
+metadata:
+  # 'name' must match the last part of the image name for container-based function.
+  # for example: the name should be set-namespace when image name is docker.example.co/functions/set-namespace
+  name: set-namespace
+spec:
+  apiVersion: example.com/v1beta1 # Primary apiVersion
+  kind: SetNamespace # Primary kind
+  description: "A short description of the KRM function"
+  publisher: example.com
+  schema: <a URL pointing to the schema of the functionConfig of apiVersion example.com/v1beta1 and kind SetNamespace>
+  version: v1.2.3
+  idempotent: true|false
+  runtime:
+    exec:
+      platforms:
+      - bin: foo-amd64-linux
+        os: linux
+        arch: amd64
+        uri: https://example.com/foo-amd64-linux.tar.gz
+        sha256: <hash>
+      - bin: foo-amd64-darwin
+        os: darwin
+        arch: amd64
+        uri: https://example.com/foo-amd64-darwin.tar.gz
+        sha256: <hash>
+  additionalAPIVersionKinds:
+    - apiVersion: example.com/v1alpha1
+      kind: LegacySetNamespace
+      deprecated: true
+  usage: <a URL pointing to a README.md>
```

Comment:
how does this work with a thing like docsify?  Docsify wants your markdown files locally AFAIK?  So will we then have to develop something that fetches the files when the index changes?

Reply:
This is a good point!

> So will we then have to develop something that fetches the files when the index changes?

I'd avoid it since it will introduce some maintenance burden.
An alternative we can consider is that we asked the contributors to check in these markdown files (usage and examples) in the registry repo and the URL in the metadata would be pointing to the markdown files in the same repo.